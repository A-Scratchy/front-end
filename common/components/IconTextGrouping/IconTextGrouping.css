.IconTextGrouping {
  align-items: center;
  display: flex;
  flex-direction: column;
  padding: 1rem;
  margin: 1rem;
  text-decoration: none;
  width: 180px;
}

.link {
  border: 1px solid transparent;
  border-radius: 4px;
  transition: border linear 0.2s, color linear 0.2s;
}

.icon {
  padding: 0.5rem 0 1rem;
}

.title {
  color: var(--slate);
  text-align: center;
}

.subtext {
  color: var(--slate);
  font-family: "Noto Serif", sans-serif;
  font-size: 0.9rem;
  text-align: left;
  width: inherit;
}

/* Theming */
.primary {
  color: var(--primary);
}

.secondary {
  color: var(--secondary);
}

.slate {
  color: var(--slate);
}

.link:hover,
.link:focus {
  border: 1px solid var(--slateLight);
  outline: none;
}

.primary.link:hover,
.primary.link:focus {
  color: var(--primaryLight);
}

.secondary.link:hover,
.secondary.link:focus {
  border: 1px solid var(--slateLight);
  color: var(--secondaryLight);
}

.slate.link:hover,
.slate.link:focus {
  border: 1px solid var(--slateLight);
  color: var(--slateLight);
}

.primary.link:active:not:hover {
  color: var(--primaryDark);
}

.secondary.link:active:not:hover {
  color: var(--secondaryDark);
}

.slate.link:active:not:hover {
  color: var(--slateDark);
}
