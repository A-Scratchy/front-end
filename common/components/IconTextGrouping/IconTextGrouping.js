/* eslint-disable react/no-danger */
import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import FontAwesomeIcon from '@fortawesome/react-fontawesome';
import OutboundLink from 'common/components/OutboundLink/OutboundLink';
import styles from './IconTextGrouping.css';

IconTextGrouping.propTypes = {
  className: PropTypes.string,
  fontAwesomeIcon: PropTypes.object.isRequired,
  isIconAboveHeading: PropTypes.bool,
  iconSize: PropTypes.string,
  subText: PropTypes.string,
  theme: PropTypes.oneOf(['primary', 'secondary', 'slate']),
  title: PropTypes.string.isRequired,
  url: PropTypes.string,
};

IconTextGrouping.defaultProps = {
  className: '',
  isIconAboveHeading: false,
  iconSize: '6x',
  subText: '',
  theme: 'slate',
  url: undefined,
};

function IconTextGrouping({
  className,
  fontAwesomeIcon,
  iconSize,
  isIconAboveHeading,
  subText,
  theme,
  title,
  url,
}) {
  const iconTextGroupingClassNames = classNames(
    styles.IconTextGrouping, className, styles[theme], { [styles.link]: !!url },
  );

  const content = (
    <React.Fragment>
      {isIconAboveHeading && (
        <FontAwesomeIcon
          className={styles.icon}
          icon={fontAwesomeIcon}
          size={iconSize}
        />
      )}
      {<h5 className={styles.title}>{title}</h5>}
      {!isIconAboveHeading && (
        <FontAwesomeIcon
          className={styles.icon}
          icon={fontAwesomeIcon}
          size={iconSize}
        />
      )}
      {subText && <div className={styles.subtext}>{subText}</div>}
    </React.Fragment>
  );

  return url ? (
    <OutboundLink
      analyticsEventLabel={`${title} <IconTextGrouping>`}
      className={iconTextGroupingClassNames}
      href={url}
      hasIcon={false}
    >
      {content}
    </OutboundLink>
  ) : (
    <div className={iconTextGroupingClassNames}>{content}</div>
  );
}

export default IconTextGrouping;
