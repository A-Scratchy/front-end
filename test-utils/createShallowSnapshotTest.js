/* eslint-env jest */
import ShallowRenderer from 'react-test-renderer/shallow';

/**
 * Used to create a snapshot test off of surface leaves of the tree.
 * Useful for regression tests on containers or larger components.
 *
 * @export
 * @param {*} Component
 */
export default Component => {
  const shallowRenderer = new ShallowRenderer();
  const tree = shallowRenderer.render(Component);

  expect(tree).toMatchSnapshot();
};
