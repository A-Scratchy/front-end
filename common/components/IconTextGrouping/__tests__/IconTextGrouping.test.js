import React from "react";
import createShallowSnapshotTest from "test-utils/createShallowSnapshotTest";

import IconTextGrouping from "../IconTextGrouping";

describe('IconTextGrouping', () => {
    test('it should render properly with all required props', () => {
        createShallowSnapshotTest(
        <IconTextGrouping
            fontAwesomeIcon='test-icon'
            title='test title'
        >
            Test
        </IconTextGrouping>
        )
    });

    test('it should render properly with some props assigned', () => {
        createShallowSnapshotTest(
            <IconTextGrouping
                className='iconClass'
                fontAwesomeIcon='test'
                isIconAboveHeading={false}
                iconSize='5x'
                subText='some test text'
                theme='slate'
                title='textTitle'
                url='www.testwebsite.com'
            >
                Test
            </IconTextGrouping>
        )
    });
});
