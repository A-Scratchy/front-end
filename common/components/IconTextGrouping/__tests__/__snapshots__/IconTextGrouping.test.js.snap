// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconTextGrouping it should render properly with all required props 1`] = `
<div
  className="IconTextGrouping slate"
>
  <React.Fragment>
    <h5
      className="title"
    >
      test title
    </h5>
    <FontAwesomeIcon
      border={false}
      className="icon"
      fixedWidth={false}
      flip={null}
      icon="test-icon"
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size="6x"
      spin={false}
      symbol={false}
      transform={null}
    />
    
  </React.Fragment>
</div>
`;

exports[`IconTextGrouping it should render properly with some props assigned 1`] = `
<withRouter(OutboundLink)
  analyticsEventLabel="textTitle <IconTextGrouping>"
  className="IconTextGrouping iconClass slate link"
  hasIcon={false}
  href="www.testwebsite.com"
>
  <React.Fragment>
    <h5
      className="title"
    >
      textTitle
    </h5>
    <FontAwesomeIcon
      border={false}
      className="icon"
      fixedWidth={false}
      flip={null}
      icon="test"
      listItem={false}
      mask={null}
      name=""
      pull={null}
      pulse={false}
      rotation={null}
      size="5x"
      spin={false}
      symbol={false}
      transform={null}
    />
    <div
      className="subtext"
    >
      some test text
    </div>
  </React.Fragment>
</withRouter(OutboundLink)>
`;
